test/SILOptimizer/existential_transform_soletype.swift

```diff
@@ -0,0 +1,52 @@
+// RUN: %target-swift-frontend -O -wmo -sil-existential-specializer -Xllvm -sil-disable-pass=GenericSpecializer -Xllvm -sil-disable-pass=FunctionSignatureOpts -Xllvm -sil-disable-pass=SILCombine -emit-sil -sil-verify-all %s | %FileCheck %s
+
+internal protocol SPP {
+  func bar()  -> Int32
+}
+internal class SCC: SPP {
+  @inline(never) func bar() -> Int32 {
+   return 5
+  }
+}
+
+@inline(never) internal func opt2(b:SPP) -> Int32{
+ return b.bar()
+}
+
+@inline(never) internal func opt1(b:SPP) -> Int32{
+ return opt2(b:b)
+}
+
+// CHECK-LABEL: sil hidden [noinline] @$s30existential_transform_soletype4opt11bs5Int32VAA3SPP_p_tF : $@convention(thin) (@in_guaranteed SPP) -> Int32 {
+// CHECK: bb0(%0 : $*SPP):
+// CHECK:   debug_value_addr
+// CHECK:   function_ref @$s30existential_transform_soletype4opt21bs5Int32VAA3SPP_p_tFTf4e_n : $@convention(thin) <τ_0_0 where τ_0_0 : SPP> (@in_guaranteed τ_0_0) -> Int32 // user: %4
+// CHECK:   open_existential_addr
+// CHECK:   apply
+// CHECK:   return
+// CHECK-LABEL: } // end sil function '$s30existential_transform_soletype4opt11bs5Int32VAA3SPP_p_tF'
+
+// CHECK-LABEL: sil shared [noinline] @$s30existential_transform_soletype4opt21bs5Int32VAA3SPP_p_tFTf4e_n : $@convention(thin) <τ_0_0 where τ_0_0 : SPP> (@in_guaranteed τ_0_0) -> Int32 {
```

Comment:
opt1 does not get optimized since the caller foo is under @_optimize(none).

Reply:
It's generally better to use `.sil` test cases for individual passes, but not an absolute requirement. The source is definitely helpful, but that can be included as a comment.